plaso/parsers/esedb_plugins/msie_webcache_cookies.py

```diff
@@ -0,0 +1,167 @@
+    flags (int): an representation of cookie flags
+    modification_time (dfdatetime.DateTimeValues): modification date and time.
+    rdomain (str): Request domain for which the cookie was set.
+  """
+
+  DATA_TYPE = 'msie:cookie:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(MsieWebCacheCookieData, self).__init__(
+        data_type=self.DATA_TYPE)
+    self.container_identifier = None
+    self.cookie_hash = None
+    self.cookie_name = None
+    self.cookie_value_raw = None
+    self.cookie_value = None
+    self.entry_identifier = None
+    self.expiration_time = None
+    self.flags = None
+    self.modification_time = None
+    self.rdomain = None
+
+class MSIE11CookiePlugin(
+    interface.ESEDBPlugin, cookie_plugins_helper.CookiePluginsHelper):
+  """SQLite parser plugin for Internet cookies in the MSIE webcache file."""
+  NAME = 'msie_webcache_cookies'
+  DATA_FORMAT = (
+      'Internet Explorer WebCache ESE database (WebCacheV01.dat, '
+      'WebCacheV24.dat) file')
+  REQUIRED_TABLES = {
+      'Containers': 'ParseContainersTableCookie',
+  }
+
+
+  def _GetDateTimeValue(self, record_values, value_name):
+    """Retrieves a date and time record value.
+
+    Args:
+      record_values (dict[str,object]): values per column name.
+      value_name (str): name of the record value.
+
+    Returns:
+      dfdatetime.DateTimeValues: date and time or None if not set.
+    """
+    filetime = record_values.get(value_name, None)
+    if not filetime:
+      return None
+
+    # TODO: add support for filetime == 1 and other edge cases.
+
+    if filetime == 0x7fffffffffffffff:
+      return dfdatetime_semantic_time.SemanticTime(string='Infinite')
+
+    return dfdatetime_filetime.Filetime(timestamp=filetime)
+
+  def _ParseCookieExTable(self, parser_mediator, table):
+    """Parses a CookieEntryEx_# table.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      table (pyesedb.table): table.
+      container_name (str): container name, which indicates the table type.
+    """
+    for record_index, esedb_record in enumerate(table.records):
+      if parser_mediator.abort:
+        break
+
+      try:
+        record_values = self._GetRecordValues(
+            parser_mediator, table.name, record_index, esedb_record)
+
+      except UnicodeDecodeError:
+        parser_mediator.ProduceExtractionWarning((
+            'Unable to retrieve record values from record: {0:d} '
+            'in table: {1:s}').format(record_index, table.name))
+        continue
+
+      def CookieHexToAscii(raw_cookie):
+        if raw_cookie is not None:
+          if raw_cookie.endswith(b'\x00'):
+            raw_cookie = raw_cookie[:-1]
+          return raw_cookie.decode('ascii')
+        return None
+      cookie_name = CookieHexToAscii(record_values.get('Name',None))
```

Comment:
style nit: use spaces between arguments

I also would recommend moving the record_values.get() into the method e..g GetRawCookieValue()